x/evm/keeper/state_transition.go

```diff
@@ -426,6 +483,8 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context, msg core.Message, trace
 		}
 	}
 
+	k.ApplyEvents(ctx, vmErr != nil)
```

Comment:
events only applied if not solidity errors, so all func execution has been succeed.